### PR TITLE
FUSETOOLS2-2077: Create UI test for Specific Camel JBang version sett…

### PR DESCRIPTION
**Pull request by pospisilf**
…ings

# Pull Request informations

## Rebase & Merge default requirements

1. Green build for main branch
2. Wait 24 hours after PR creation
3. Green job for PR
4. Approved PR

## PR labels default process

- READY_FOR_REVIEW  &rightarrow;  REVIEW_DONE  &rightarrow;  READY_FOR_MERGE

## Tests

- [ ] Are there **Unit tests**?
- [ ] Are there **Integration tests**?
- [ ] Do we need a new **UI test**?

## PR workflow progress

1. [ ] Tagged with relevant **PR labels**
2. [ ] Green **job for PR**
3. [ ] PR was created more than **24 hours ago** or **All committers approved** it
4. [ ] Green **main** branch build

### CODE REVIEW

**Review by djelinek — CHANGES_REQUESTED** [review]

File: src/ui-test/tests/jbang_user_preference.test.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        const textField = await (await new Workbench().openSettings()).findSetting('JBang Version', 'Camel', 'Language Support');
+        await textField.setValue(version);
+        await driver.sleep(500);
+        await new EditorView().closeEditor('Settings');
+    }
+
+    async function waitUntilTerminalHasText(driver: WebDriver, textArray: string[], interval = 500): Promise<void> {
+        await driver.wait(async function () {
+            try {
+                await DefaultWait.sleep(10000);
```
Comment (pospisilf) [inline src/ui-test/tests/jbang_user_preference.test.ts:134]: unstable if no static wait here - do not know why yet

Comment (pospisilf) [reply src/ui-test/tests/jbang_user_preference.test.ts]: huge thanks to @djelinek for finding solution

File: src/ui-test/tests/jbang_user_preference.test.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    }
+
+    async function activateTerminalView(): Promise<TerminalView> {
+        // workaround ExTester issue - https://github.com/redhat-developer/vscode-extension-tester/issues/785
+        await new Workbench().executeCommand('Terminal: Focus on Terminal View');
+        return await new BottomBarPanel().openTerminalView();
+    }
+
+    function deleteFile(filename: string): void {
+        fs.remove(path.resolve(RESOURCES, filename), (err: any) => {
```
Comment (djelinek) [inline src/ui-test/tests/jbang_user_preference.test.ts:163]: `remove` is async.. you need to modify method to return Promise and use await or use `fs.removeSync` instead

File: src/ui-test/tests/jbang_user_preference.test.ts [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+            await input.confirm();
+
+            await driver.wait(async function () {
+                return (await new EditorView().getOpenEditorTitles()).find(t => t === FILENAME);
+            }, 220000);
+
+            await waitUntilTerminalHasText(driver, [`-Dcamel.jbang.version=${DEFAULT_JBANG}`]);
+            expect(await (await activateTerminalView()).getText()).to.contain(`-Dcamel.jbang.version=${DEFAULT_JBANG}`);
+
+            terminalView = await new BottomBarPanel().openTerminalView();
```
Comment (djelinek) [inline src/ui-test/tests/jbang_user_preference.test.ts:81]: terminal cleanup could be moved to `afterEach`

File: src/ui-test/tests/jbang_user_preference.test.ts [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    describe('Different JBang versions', function () {
+        afterEach(async function () {
+            await new EditorView().closeAllEditors();
+            deleteFile(FILENAME);
+        });
+
+        it('Default version', async function () {
+            await setJBangVersion(DEFAULT_JBANG);
+
+            await new Workbench().executeCommand('Camel: Create a Camel Route using XML DSL');
```
Comment (djelinek) [inline src/ui-test/tests/jbang_user_preference.test.ts:65]: init and wait for file is created could be moved to method and reused in both tests

File: src/ui-test/tests/jbang_user_preference.test.ts [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+ * limitations under the License.
+ */
+
+import { expect } from 'chai';
+import path = require('path');
+import { WebDriver, VSBrowser, Marketplace, By, EditorView, Workbench, InputBox, BottomBarPanel, TerminalView } from 'vscode-uitests-tooling';
+import * as pjson from '../../../package.json';
+import * as fs from 'fs-extra';
+
+describe('JBang user preference version set test', function () {
```
Comment (djelinek) [inline src/ui-test/tests/jbang_user_preference.test.ts:24]: I would maybe change the filename from `jbang_user_preference.test.ts` to `camel_jbang_version.test.ts` it seems to me more precise and it goes hand to hand with already existing `camel_catalog_version.test.ts` WDYT?

**Review by djelinek — CHANGES_REQUESTED** [review]

**Review by djelinek — APPROVED** [review]

**Pull request merged**